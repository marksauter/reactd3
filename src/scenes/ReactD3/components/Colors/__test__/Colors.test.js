import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import ConnectedColors, { Colors } from '../';
import { reverseSwatches } from '../';
import { initialState } from '../';

const mockStore = configureStore();

// Snapshot for Colors
describe('Colors --- Snapshot', () => {
  it('matches snapshot', () => {
    const renderedValue = renderer.create(<Colors width="600" />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
//***********************************************************************
describe('Colors --- shallow render', () => {
  const width = 600;
  const wrapper = shallow(<Colors width={width} />);
  
  it('renders component', () => {
    expect(wrapper.length).toEqual(1);
  });
});
//***********************************************************************
describe('Colors --- shallow + mock store', () => {
  const store = mockStore({ colors: initialState });
  const wrapper = shallow(<ConnectedColors width="600" store={store} />);

  it('renders connected component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('state mapped to props', () => {
    expect(wrapper.prop('reversed')).toEqual(initialState.reversed);
  });
});
//***********************************************************************
describe('Colors --- mount + Provider', () => {
  const store = mockStore({ colors: initialState });
  const wrapper = mount(
    <Provider store={store}>
      <ConnectedColors width="600" />
    </Provider>
  );

  it('renders connected component', () => {
    expect(wrapper.find(ConnectedColors).length).toEqual(1);
  });

  it('state mapped to props', () => {
    expect(wrapper.find(Colors).prop('reversed')).toEqual(initialState.reversed);
  });
});
