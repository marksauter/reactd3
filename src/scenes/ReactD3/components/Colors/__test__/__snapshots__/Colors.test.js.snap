// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors --- Snapshot matches snapshot 1`] = `
<g
  onClick={[Function]}
>
  <rect
    height="20"
    style={
      Object {
        "fill": "#1f77b4",
      }
    }
    width={30}
    x={570}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#aec7e8",
      }
    }
    width={30}
    x={540}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#ff7f0e",
      }
    }
    width={30}
    x={510}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#ffbb78",
      }
    }
    width={30}
    x={480}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#2ca02c",
      }
    }
    width={30}
    x={450}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#98df8a",
      }
    }
    width={30}
    x={420}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#d62728",
      }
    }
    width={30}
    x={390}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#ff9896",
      }
    }
    width={30}
    x={360}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#9467bd",
      }
    }
    width={30}
    x={330}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#c5b0d5",
      }
    }
    width={30}
    x={300}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#8c564b",
      }
    }
    width={30}
    x={270}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#c49c94",
      }
    }
    width={30}
    x={240}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#e377c2",
      }
    }
    width={30}
    x={210}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#f7b6d2",
      }
    }
    width={30}
    x={180}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#7f7f7f",
      }
    }
    width={30}
    x={150}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#c7c7c7",
      }
    }
    width={30}
    x={120}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#bcbd22",
      }
    }
    width={30}
    x={90}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#dbdb8d",
      }
    }
    width={30}
    x={60}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#17becf",
      }
    }
    width={30}
    x={30}
  />
  <rect
    height="20"
    style={
      Object {
        "fill": "#9edae5",
      }
    }
    width={30}
    x={0}
  />
</g>
`;
